src/Mapping/ExtensionMetadataFactory.php

```diff
@@ -114,7 +114,7 @@ public function getExtensionMetadata($meta)
         if (null !== $meta->reflClass) {
             foreach (array_reverse(class_parents($meta->getName())) as $parentClass) {
                 // read only inherited mapped classes
-                if ($cmf->hasMetadataFor($parentClass)) {
+                if ($cmf->hasMetadataFor($parentClass) || !$cmf->isTransient($parentClass)) {
```

Comment:
I'm wondering if the `$cmf->hasMetadataFor($parentClass) || ` part is necessary since it's confusing what it does and not reliable.

Reply:
IMO it can stay as is because the `hasMetadataFor` call is doing just an `isset` check under the hood while the `isTransient` call is more expensive since it goes to the mapping driver which then has to check whether the class with the specified name should have its metadata loaded  or not (for example in the case of the atrribute driver it has to use reflection to check whether the class is mapped as an entity or as a mapped super class or neither).

Reply:
yeah, I guess it would save some calls to the driver, looking at the implementation of `getParentClasses`, they only use `isTransient` 🤔 :
https://github.com/doctrine/persistence/blob/413eb71a22c31c309b5a1cea9701d723cc0d4ae2/src/Persistence/Mapping/AbstractClassMetadataFactory.php#L278-L292